git_machete/tests/functional/test_machete.py

```diff
@@ -25,6 +28,170 @@ def get_head_commit_hash() -> str:
+        self.opt_override_to_parent: bool = False
+        self.opt_push_tracked: Optional[bool] = True
+        self.opt_push_untracked: Optional[bool] = True
+        self.opt_return_to: str = "stay"
+        self.opt_roots: List[str] = list()
+        self.opt_start_from: str = "here"
+        self.opt_stat: bool = False
+        self.opt_sync_github_prs: bool = False
+        self.opt_unset_override: bool = False
+        self.opt_yes: bool = True
+
+
+class GitAPIState:
+    pulls: List[Dict[str, Any]] = [{'head': {'ref': 'bugfix/remove'}, 'user': {'login': 'github_user'}, 'base': {'ref': 'develop'}, 'number': '15', 'html_url': 'www.github.com'}]
+    user: Dict[str, str] = {'login': 'github_user', 'type': 'User', 'company': 'VirtusLab'}
+    issues: List[Dict[str, Any]] = []
+    remote_branches: List[str] = []
+
+    def __init__(self, url: str, headers: Dict[str, str] = None, data: Union[str, bytes, None] = None, method: str = ''):
+        self.url: str = url
+        self.status_code: Optional[int] = None
+        self.json_data: Union[str, bytes] = data
+        self.return_data: Optional[Union[List[Dict[str, Any]], Dict[str, Any]]] = None
+        self.headers: Dict[str, str] = headers
+        if method == "GET":
+            self.handle_get()
+        elif method == "PATCH":
+            self.handle_patch()
+        elif method == "POST":
+            self.handle_post()
+
+    def read(self) -> Union[str, bytes]:
+        return json.dumps(self.return_data).encode()
+
+    def handle_get(self) -> None:
+        if 'pulls' in self.url:
+            if '?head' in self.url:
+                head: str = self.url[self.url.rfind(':') + 1:]
+                for pr in GitAPIState.pulls:
+                    if pr['head']['ref'] == head:
+                        self.return_data = [pr]
+                        self.status_code = 200
+                        return
+                self.return_data = []
+                self.status_code = 404
+                return
+            else:
+                self.status_code = 200
+                self.return_data = GitAPIState.pulls
+                return
+        elif 'user' in self.url:
+            self.status_code = 200
+            self.return_data = GitAPIState.user
+            return
+
+    def handle_patch(self) -> None:
+        if 'issues' in self.url:
+            self.update_issue()
+        elif 'pulls' in self.url:
+            self.update_pull()
+
+    def handle_post(self) -> None:
+        if 'issues' in self.url:
+            if self.url.endswith('issues'):
+                self.update_issue(new=True)
+        elif 'pulls' in self.url:
+            if self.url.endswith('pulls'):
+                self.update_pull(new=True)
+
+    def update_pull(self, new: bool = False) -> None:
+        pull: Dict[str, Any]
+        if new:
+            pull = {'number': self.get_new_number(GitAPIState.pulls),
+                    'user': {'login': 'githubuser'},
```

Comment:
Minor: I would rename it slightly, because some people when they read code they're paranoid (see @me) and they're read such places in case someone decided to put a crucial information like `login:not_really_login_actually_a_password`, because of some dirty hack somewhere else. With _ You can read it faster and at a first glance understand what it says. As a plus it will stop looking like `git abuser` 🗡️ 

```suggestion
                    'user': {'login': 'github_user'},
```